main.py

```diff
@@ -0,0 +1,47 @@
+import telebot
+import pyqrcode
+from PIL import Image
+from pyzbar.pyzbar import decode
+
+bot = telebot.TeleBot('6046831990:AAGMjVSRxnL_WX9IdfVcMLgjyoUcIjmSBYg')
+
+qr_code_filename = 'qr_code.png'
+
+
+def download_file(file_id: str) -> None:
+    qr_code_info = bot.get_file(file_id)
+    qr_code_file = bot.download_file(qr_code_info.file_path)
+
+    with open(qr_code_filename, 'wb') as file:
+        file.write(qr_code_file)
+
+
+@bot.message_handler(commands=['start', 'help'])
+def starter(message):
+    bot.send_message(message.chat.id, "Hello, send me the text and I will generate a qr code")
+
+
+@bot.message_handler(content_types=['text'])
+def text_to_qr(message):
+    qr_code = pyqrcode.create(message.text)
+    qr_code.png(qr_code_filename, scale=6)
+
+    with open(qr_code_filename, 'rb') as photo:
+        bot.send_photo(message.chat.id, photo)
+
+
+@bot.message_handler(content_types=['photo'])
+def qr_to_text(message):
+    photo_id = message.photo[-1].file_id
+    download_file(photo_id)
```

Comment:
try to download the file to memory and pass to pillow, without saving in filesystem